tensorflow/core/util/mkl_util.h

```diff
@@ -663,6 +869,24 @@ inline void AllocTmpBuffer(OpKernelContext* context, Tensor* tensor_out,
                                                  tf_shape, tensor_out));
 }
 
+#ifdef ENABLE_MKLDNN_V1
+inline void GetStridesFromSizes(MklTensorFormat data_format, size_t* strides,
```

Comment:
At the beginning of this file, we could define
```C++
#ifdef ENABLE_MKLDNN_V1
#define MKL_TENSOR_FORMAT MklTensorFormat
#define MKL_TENSOR_FORMAT_NHWC MklTensorFormat::FORMAT_NHWC
#else 
#define MKL_TENSOR_FORMAT TensorFormat
#define MKL_TENSOR_FORMAT_NHWC FORMAT_NHWC
#endif  // ENABLE_MKLDNN_V1
```
(Then undefine all these macros at the end of file.)

And at functions like this, we can just do
```C++
inline void GetStridesFromSizes(MKL_TENSOR_FORMAT data_format, size_t* strides,
                                const size_t* sizes) {
#ifdef ENABLE_MKLDNN_V1
  DCHECK_NE(data_format, MklTensorFormat::FORMAT_UNDEF);
#endif  // ENABLE_MKLDNN_V1
  // MKL requires strides in NCHW
  if (data_format == MKL_TENSOR_FORMAT_NHWC) {
    strides[0] = sizes[2];
    strides[1] = sizes[0] * sizes[2];
    strides[2] = 1;
    strides[3] = sizes[0] * sizes[1] * sizes[2];
  } else {
    strides[0] = 1;
    strides[1] = sizes[0];
    strides[2] = sizes[0] * sizes[1];
    strides[3] = sizes[0] * sizes[1] * sizes[2];
  }
}
```
This could save a lot of space and makes maintaining easier (less redundant code). Please try to do this with as many functions as possible.

Reply:
Agree. Done